stock-db/stock_db_solution.py

```diff
@@ -0,0 +1,140 @@
+
+cur = conn.cursor()
+
+query1 = cur.execute("SELECT name FROM sqlite_master WHERE type='table';")
+tables = query1.fetchall()
+
+tables # STOCK_DATA is the table name
+
+query2 = cur.execute("SELECT * FROM STOCK_DATA")
+data = query2.fetchall()
+data
+
+stock_data_cols_query = cur.execute("PRAGMA table_info(STOCK_DATA)") # Getting column names of STOCK_DATA
+stock_data_cols = [i[1] for i in stock_data_cols_query.fetchall()]
+stock_data_cols
+
+# Create table where Symbol = 'MSFT'
+create_msft_table = cur.execute("CREATE TABLE msft_table(`index`, Date, Open, High, Low, Close, Volume, Adj Close, Symbol);")
+
+# For msft_table
+msft_query = cur.execute("""
+                SELECT *
+                FROM STOCK_DATA
+                WHERE Symbol = 'MSFT';
+              """)
+msft_list = msft_query.fetchall()
+
+cur.executemany("INSERT INTO msft_table VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)", msft_list)
+
+msft_table_query = cur.execute("SELECT * FROM msft_table")
+msft_table_list = msft_table_query.fetchall()
+msft_table_list
+
+# Create table where Symbol = AAPL
+create_aapl_table = cur.execute("CREATE TABLE aapl_table(`index`, Date, Open, High, Low, Close, Volume, Adj Close, Symbol);")
+
+# For appl_table
+aapl_query = cur.execute("""
+                SELECT *
+                FROM STOCK_DATA
+                WHERE Symbol = 'AAPL';
+              """)
+aapl_list = aapl_query.fetchall()
+
+cur.executemany("INSERT INTO aapl_table VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)", aapl_list)
+
+aapl_table_query = cur.execute("SELECT * FROM aapl_table")
+aapl_table_list = aapl_table_query.fetchall()
+aapl_table_list
+
+"""# **#2 Read the two new new tables in from the database using SQL to check if they were successfully created.**"""
+
+msft_table_list
+
+aapl_table_list
+
+"""# **#3 For each new table in the database, query for the Maximum and Minimum dates, and save those as new pandas data frames.**"""
+
+# I was unsure if you wanted the two dates as their own dataframe or if you
+# wanted all rows where the dates are equal to either of those.
+# I went ahead and made two separate dataframes for each anyway
+
+
+# msft_table max and min dates
+
+msft_max_min_query = cur.execute("""
+                                  SELECT MAX(Date) as max_date, MIN(Date) as min_date
+                                  FROM msft_table;
+                                  """)
+msft_max_min_list = msft_max_min_query.fetchall()
+msft_max_min_list # max_date is '2014-07-21' and min_date is '2000-01-03'
+
+msft_max_min_df = pd.DataFrame(data = msft_max_min_list)
+msft_max_min_df
+
+msft_max_min_table_query = cur.execute("""
+                            SELECT *
+                            FROM msft_table
+                            WHERE Date = '2014-07-21' OR Date = '2000-01-03';
+                            """)
+msft_max_min_table_list = msft_max_min_table_query.fetchall()
+msft_max_min_table_df = pd.DataFrame(data = msft_max_min_table_list)
```

Comment:
Don't forget about column names (hint: you can also use pd.read_sql() to automatically get the column names). This applies to the rest of the dataframes below as well.